File: shuttle-api/src/ShuttleApi.Domain/Clients/ContractRateLine.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Clients;

public sealed class ContractRateLine : Entity<Guid>
{
    public Guid ContractId { get; private set; }
    public string BillingCode { get; private set; } = string.Empty;
    public string Description { get; private set; } = string.Empty;
    public string VehicleType { get; private set; } = string.Empty;
    public int? MaxDistanceKm { get; private set; }
    public bool CargoIncluded { get; private set; }
    public decimal DayRate { get; private set; }

    private ContractRateLine() { }

    public static ContractRateLine Create(
        Guid contractId,
        string billingCode,
        string description,
        string vehicleType,
        int? maxDistanceKm,
        bool cargoIncluded,
        decimal dayRate)
    {
        return new ContractRateLine
        {
            Id = Guid.NewGuid(),
            ContractId = contractId,
            BillingCode = billingCode,
            Description = description,
            VehicleType = vehicleType,
            MaxDistanceKm = maxDistanceKm,
            CargoIncluded = cargoIncluded,
            DayRate = dayRate
        };
    }
}

File: shuttle_tablet/lib/features/clients/domain/entities/contract_rate_line.dart
import 'package:equatable/equatable.dart';

class ContractRateLine extends Equatable {
  final String id;
  final String contractId;
  final String billingCode;
  final String description;
  final String vehicleType;
  final int? maxDistanceKm;
  final bool cargoIncluded;
  final double dayRate;

  const ContractRateLine({
    required this.id,
    required this.contractId,
    required this.billingCode,
    required this.description,
    required this.vehicleType,
    this.maxDistanceKm,
    required this.cargoIncluded,
    required this.dayRate,
  });

  @override
  List<Object?> get props => [id, contractId, billingCode, description, vehicleType, maxDistanceKm, cargoIncluded, dayRate];
}
